src/main/java/com/thullo/web/controller/BoardController.java

```diff
@@ -27,13 +29,32 @@
 @RequiredArgsConstructor
 public class BoardController {
     private final BoardService boardService;
-    @PostMapping
-    public ResponseEntity<ApiResponse> createBoard(@RequestParam(value = "file", required = false) MultipartFile file, @RequestParam("boardName") String boardName,
-                                                   @CurrentUser UserPrincipal principal, HttpServletRequest request) {
+    @PostMapping(consumes = {MediaType.MULTIPART_FORM_DATA_VALUE}, produces = {MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity<ApiResponse> createBoard(@RequestParam(value = "file", required = false) MultipartFile file,
+                                                   @RequestParam(value = "boardName") String boardName, @CurrentUser UserPrincipal principal,
+                                                   @RequestParam(value = "boardVisibility", required = false) String boardVisibility, HttpServletRequest request) {
         try {
-            BoardRequest boardRequest = new BoardRequest(boardName, request.getRequestURL().toString(), file);
+            BoardRequest boardRequest = new BoardRequest(boardName, request.getRequestURL().toString(), file, boardVisibility);
             BoardResponse board = boardService.createBoard(boardRequest, principal);
             return ResponseEntity.ok(new ApiResponse(true, "Board successfully created", board));
+
+        } catch (UserException | IOException | BadRequestException ex) {
+            return ResponseEntity.badRequest().body(new ApiResponse(false, ex.getMessage(),
+                    new HashMap<>(Map.of("message", ex.getMessage()))));
+        }
+    }
+
+    @PutMapping(value = "/{boardTag}", consumes = {MediaType.MULTIPART_FORM_DATA_VALUE}, produces = {MediaType.APPLICATION_JSON_VALUE})
+    @PreAuthorize("@boardServiceImpl.hasBoardRole(authentication.principal.email, #boardTag) or hasRole('BOARD_' + #boardTag)")
+    public ResponseEntity<ApiResponse> updateBoard(@PathVariable String boardTag,
```

Comment:
I recall instructing you to add a task on the board for refactoring the createBoardController. this will require an update